source/dlangbot/bugzilla.d

```diff
@@ -37,9 +37,42 @@ unittest
                  [IssueRef(16319, true)]));
     assert(equal(matchIssueRefs("Fixes issues 17494, 17505, 17506"),
                  [IssueRef(17494, true),IssueRef(17505, true), IssueRef(17506, true)]));
+    assert(equal(matchIssueRefs("Fix issues 42, 55, 98: Baguette poisson fraise"),
+                 [ IssueRef(42, true), IssueRef(55, true), IssueRef(98, true)  ]));
     // only first match considered, see #175
     assert(equal(matchIssueRefs("Fixes Issues 1234 and 2345\nblabla\nFixes Issue 3456"),
                  [IssueRef(1234, true), IssueRef(2345, true)]));
+    // Related, but not closing
+    assert(equal(matchIssueRefs("Issue 242: Refactor prior to fix"),
+                 [IssueRef(242, false)]));
+    assert(equal(matchIssueRefs("Bug 123: Add a test"),
+                 [IssueRef(123, false)]));
+    assert(equal(matchIssueRefs("Issue #456: Improve error message"),
+                 [IssueRef(456, false)]));
+
+    // Short hand syntax
+    assert(equal(matchIssueRefs("Fix 222, 333 and 42000: Baguette poisson fraise"),
+                 [ IssueRef(222, true), IssueRef(333, true), IssueRef(42000, true)  ]));
+    assert(equal(matchIssueRefs("Fix 4242 & 131 Baguette poisson fraise"),
+                 [ IssueRef(4242, true), IssueRef(131, true)  ]));
+    // Just a reference, not a fix
+    assert(equal(matchIssueRefs("Issue 242: Warn about buggy behavior"),
+                 [IssueRef(242, false)]));
+    assert(equal(matchIssueRefs("Do not quite fix issue 242 but it's a start"),
+                 [IssueRef(242, false)]));
+    assert(equal(matchIssueRefs("Workaround needed to make bug 131415 less deadly"),
+                 [IssueRef(131415, false)]));
+
+    // Shouldn't match
+    const IssueRef[] empty;
+    assert(equal(matchIssueRefs("Will fix 242 and 131 later"), empty));
```

Comment:
Shouldn't this be a non-fix reference?

Reply:
No because there's no "issue" or "bug" before. Matching numbers would lead to way too many false positive, I fear (e.g. `32 bits`).

Reply:
Fair enough.